File: starter-bots/ReferenceBot/Domain/Models/BotStateDTO.cs
namespace Domain.Models
{
  public class BotStateDTO
  {
    public int CurrentLevel { get; set; }
    public string ConnectionId { get; set; }
    public int Collected { get; set; }
    public string ElapsedTime { get; set; }
    public int[][] HeroWindow { get; set; }
    public int X { get; set; }
    public int Y { get; set; }
  }
}

File: starter-bots/KotlinBot/src/main/kotlin/models/dtos/BotStateDto.kt
package models.dtos

class BotStateDto(var x: Int, var y: Int, var connectionId: String, var elapsedTime: String, var collected: Int, var heroWindow: Array<IntArray>?, var currentLevel: Int) {

    override fun toString(): String {
        val windowStr = StringBuilder()
        if (heroWindow != null) {
            for (y in heroWindow!![0].indices.reversed()) {
                for (x in heroWindow!!.indices) {
                    windowStr.append(heroWindow!![x][y])
                }
                windowStr.append("\n")
            }
        }
        return String.format("Position: (%d, %d), Level: %d, Collected: %d\n%s", x, y, currentLevel, collected, windowStr.toString())
    }
}
